internal/controller/dnsrecord_controller.go

```diff
@@ -225,6 +225,8 @@ func (r *DNSRecordReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		return r.updateStatus(ctx, previous, dnsRecord, false, err)
 	}
 
+	dnsRecord.SetStatusGroup(r.Group)
```

Comment:
This is causing me a bit of a headache, if this is empty in the remote reconciler, I don't know if that's because the remote has no group set, or because it hasn't set it in the status yet.

Reply:
The remote reconcile won't proceed until the local one has added the owner and group and set the "ReadyForDelegation" status condition. Not really seeing the problem, if it's empty then it's because it has no group set by the local controller. I asked a question about this on here https://github.com/Kuadrant/dns-operator/pull/632#pullrequestreview-3453814300 but got no reply. Only makes sense to me that the local controller is setting the group though, so that's what this is doing.